lib/AST.h

```diff
@@ -61,6 +61,7 @@ namespace pillar
       const std::shared_ptr<mlir::Operation> module;
       const mlir::DataLayout dl;
       const NameProvider np;
+      static std::vector<std::function<void(void)>> *lift_queue;
```

Comment:
You don't need to make it `static` or heap-allocated either, just `std::vector<std::function<void(void)>> lift_queue;` will do.

Reply:
we use lift_queue in a static member function CreateFromModule, this was the only way I was able to use it.